server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -1080,7 +1081,11 @@ public Query wildcardQuery(
                     value = indexedValueForSearch(value).utf8ToString();
                 }
 
-                if (caseInsensitive == false && storedInBinaryDocValues() == false) {
+                if (storedInBinaryDocValues()) {
+                    return new SlowCustomBinaryDocValuesWildcardQuery(name(), value, caseInsensitive);
+                }
+
+                if (caseInsensitive == false) {
```

Comment:
question: why not using `SlowCustomBinaryDocValuesWildcardQuery` when case sensitive?

Reply:
I am asking this since I see in `buildByteRunAutomaton` the automaton is built based on the `caseInsensitive` flag. This pattern could potentially be applied to the `SortedSetDocValues` path as well resulting in just 2 query classes (one per doc values format), each taking a `caseInsensitive` boolean, rather than 4 separate code paths. But maybe this is what you are planning to do as a followup.